src/message/MessageListContainer.js

```diff
@@ -0,0 +1,75 @@
+import React, { PureComponent } from 'react';
+
+import { connect } from 'react-redux';
+import { connectActionSheet } from '@expo/react-native-action-sheet';
+
+import boundActions from '../boundActions';
+import MessageList from './MessageList';
+import {
+  getAuth,
+  getFlags,
+  getCurrentTypingUsers,
+  getShownMessagesInActiveNarrow,
+  getActiveNarrow,
+} from '../selectors';
+import { filterUnreadMessageIds } from '../utils/unread';
+import { registerAppActivity } from '../utils/activity';
+import { queueMarkAsRead } from '../api';
+
+class MessageListContainer extends PureComponent {
+  handleMessageListScroll = (e: Object) => {
```

Comment:
hmm...here I wanted to prevent MessageList from being dependent on `flags` props. Since it would be changing quite frequently, we'd just be updating the container and not the list itself.